### PR TITLE
Enhance unbind and deprovision test cases

### PR DESCRIPTION
**Pull request by leonwanghui**
Signed-off-by: leonwanghui <wanghui71leon@gmail.com>

This patch is first part of #66 , focusing on unbind and deprovision scenario.

### CODE REVIEW

File: 2.13/mocks/mockOSB.js [hunk 1]
```diff
@@ -187,11 +200,32 @@ app.delete('/v2/service_instances/:instance_id', function (req, res) {
...
   } else {
-    res.status(202).send({
-      'operation': 'task_10'
-    })
+    for (var i in serviceInstances) {
+      if (serviceInstances[i] === found) {
+        delete serviceInstances[i]
+      }
+    }
+    // Deprovision operation will also remove all binding info related to the instance
```
Comment (zhongyi-zhang) [inline 2.13/mocks/mockOSB.js]: I think the behavior below is not really for "all" bindings. It depends on the current mock config. The behavior is okay as a mock. But the comment should not be misleading.

File: 2.13/mocks/mockOSB.js [hunk 2]
```diff
@@ -171,9 +171,22 @@ app.delete('/v2/service_instances/:instance_id/service_bindings/:binding_id', fu
...
+    res.status(410).send({})
+    return
   } else {
-    res.status(200).send({})
+    var found = findWhichContains(serviceBindings, 'binding_id', req.params.binding_id)
+    for (var i in serviceBindings) {
+      if (serviceBindings[i] === found) {
+        delete serviceBindings[i]
+      }
+    }
```
[the remark this answers is not in the document]
Comment (leonwanghui) [reply 2.13/mocks/mockOSB.js]: Got it, thanks!

**Comment by leonwanghui** [comment]
@norshtein @mattmcneeney One more please

File: 2.13/mocks/mockOSB.js [hunk 3]
```diff
@@ -187,11 +199,31 @@ app.delete('/v2/service_instances/:instance_id', function (req, res) {
...
+    res.status(410).send({})
+    return
+  }
+  var i = serviceInstances.indexOf(found)
+  if (i > -1) {
+    delete serviceInstances[i]
+  }
+
+  // Although the spec doesn't fomulate how to deal with related binding resource when deleting
+  // service instances, the mock server would remove all binding info related to the instance.
```
Comment (norshtein) [inline 2.13/mocks/mockOSB.js]: According to Zhongyi's [comment](https://github.com/openservicebrokerapi/osb-checker/pull/69#discussion_r236111194), you may want to stop saying `all binding info` or make below statements in a loop, and I prefer the latter one.

Comment (norshtein) [reply 2.13/mocks/mockOSB.js]: maybe following code can help 
```javascript
do {
    found = findWhichContains(serviceBindings, 'instance_id', req.params.instance_id)
    var k = serviceBindings.indexOf(found)
    if (k > -1) {
      delete serviceBindings[k]
    }
  } while (k > -1);
```

Comment (leonwanghui) [reply 2.13/mocks/mockOSB.js]: Got it, thanks!

**Comment by leonwanghui** [comment]
@zhongyi-zhang @norshtein Please take a look at this patch, thanks!

**Comment by norshtein** [comment]
LGTM

**Pull request merged**